src/Concerns/AsVersionedContent.php

```diff
@@ -41,4 +51,53 @@ public function getTable()
 
         return $table;
     }
+
+    /**
+     * @param  callable(\Illuminate\Database\Eloquent\Query $query)  $callback
+     */
+    public static function withVersionActive(string|VersionNumber|VersionContract $version, Closure $callback): mixed
+    {
+        /** @var ManagesVersions $versions */
+        $versions = app(ManagesVersions::class);
+
+        $active = $versions->active();
+
+        $versions->setActive($version);
+
+        $result = $callback(static::query());
+
+        $versions->setActive($active);
+
+        return $result;
+    }
+
+    public function visibility(): Attribute
+    {
+        return Attribute::make(
+            get: function () {
+                return Revision::collection($this->visibleHistory());
+            })->shouldCache();
+    }
+
+    public function visibleHistory(): Collection
+    {
+        return $this->history()
+            ->with('version')
+            ->get()
+            ->groupBy('version_id')
+            ->map(function (Collection $items) {
+                $item = $items->sortByDesc('created_at')->first();
+
+                $trackable = static::withVersionActive($item->version, function (Builder $query) use ($item) {
+                    return $query->withoutGlobalScopes()
+                        ->whereKey($item->trackable_id)
+                        ->first();
+                });
+
+                $item->setRelation('trackable', $trackable);
+
+                return $item;
+            })
```

Comment:
This block feels tough to parse, is there any way it could be simpler? maybe just some comments explaining what's going on might help?

Reply:
Will refactor. Good call.

Reply:
Comments added.